src/Repository/Introduction.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Friendica\Repository;
+
+use Friendica\Collection;
+use Friendica\Core\Protocol;
+use Friendica\Model;
+
+
+use Friendica\Protocol\Diaspora;
+use Friendica\Repository;
+use Friendica\Util\DateTimeFormat;
+
+/**
+ * @method Model\Introduction       selectFirst(array $condition)
+ * @method Collection\Introductions select(array $condition = [], array $params = [])
```

Comment:
Again my question (sorry ^^), why don't you use it this way:
```suggestion
 * @method Collection\Introduction[] select(array $condition = [], array $params = [])
```
which would making the `Collection` class obsolete... The only thing which extends it is the `$totalCount`, which could get overriden if `Model` implements https://www.php.net/manual/de/countable.count.php and now you could use:
```php
$totalCount = 0;
foreach ($models as $model) {
   $totalCount += count($model);
}
```
in cases you need it.

As said, we would make one class-hierachy parallel to the model-classes completely obsolet...
```

Reply:
No, the `totalCount` isn't the number of models in the array/collection, I've already implemented the `Countable` interface for this. It counts the total number of matching models before pagination limit.

The larger point is that I don't want arrays for structured data. I don't want anymore `array` parameter when we know it should hold a specific type of object, and PHP doesn't enforce `Class[] $objects` in the function prototype that I know of.

Reply:
I understand, but we should really care about simple examples and a good readme/doc for new contributors about the "how" and the "why" ..

So it isn't necessary to reverse engineer the whole patterns to find a suitable way like creating new entities, factories, repos, models or use them the right way..

Reply:
Fair enough.